app/controllers/comment_controller.php

```diff
@@ -63,26 +66,103 @@ public function edit()
         $title = $thread->title;
         $page = Param::get('page_next', 'edit');
 
-
         switch ($page) {
-        case 'edit':
-            break;
-        case 'edit_end':
-           $comment->body = Param::get('body');
-
-            try {
-                $comment->edit($comment_id);
-                redirect("/comment/view?thread_id={$_SESSION['thread_id']}");
-            } catch (ValidationException $e) {
-                $page = 'edit';
-            }
-            break;  
-        default:
-            throw new NotFoundException("{$page} not found");
-            break;
+            case 'edit':
+                break;
+            case 'edit_end':
+               $comment->body = Param::get('body');
+
+                try {
+                    $comment->edit($comment_id);
+                    redirect("/comment/view?thread_id={$_SESSION['thread_id']}");
+                } catch (ValidationException $e) {
+                    $page = 'edit';
+                }
+                break;  
+            default:
+                throw new NotFoundException("{$page} not found");
+                break;
         }
 
         $this->set(get_defined_vars());
         $this->render($page);
     }
+
+    public function delete()
+    {
+        $comments = new Comment();
+
+        $comment_id = Param::get('comment_id');
+
+        $comments->deleteComment($comment_id);
+        $comments->deleteExisting($comment_id); //deletes records in like_monitor table when a comment is deleted
+
+        $this->set(get_defined_vars());
+        redirect("/comment/view?thread_id={$_SESSION['thread_id']}");
+    }
+
+    public function liked()
+    {
+        $comment = new Comment();
+
+        $comment_id = Param::get('comment_id');
+        $like_checker = $comment->hasLiked($comment_id);
+        $dislike_checker = $comment->hasDisliked($comment_id);
+
+        if (!$like_checker) { //comment has not been liked
+            if (!$dislike_checker) { //comment has not been disliked
+                $comment->addLike($comment_id); //add 1 liked record in like_monitor table
+                $comment->updateLikedCount($comment_id); //increment liked column in comment table
```

Comment:
Is it possible to just put the same logic under 1 function vs using 2 separate functions (addLike, updateLikedCount)? I think if 1 function fails, the other should not proceed as well po :bow: